src/base/path.h

```diff
@@ -35,7 +35,9 @@
 
 #include "pathfwd.h"
 
-class Path final
+#include "base/settingsstorage.h"
+
+class Path final : public IHumanEditableSetting
```

Comment:
The data type (in other words, the interface of the class) should describe its essence, its capabilities and public behavior, i.e. what this type can do, and not what you are going to use it for in a particular case. A good example is the basic interfaces used in programming languages that have a well-developed interface-based design culture (take at least C#): ICopyable, ISerializable, IIterable, etc. This allows you to form the right dependencies (and avoid cross dependencies).
The one more thing is that base and child class always have the following dependency: Child **is a** Base (e.g. Truck is a Car). But `Path is a HumanEditableSetting` is incorrect.
A path is something that can inherently be represented as a string, so that's exactly the interface it can have. In fact, the provided interface have suitable method name so now the problem is only in the naming of the interface itself and in the place of its declaration. It should be located in a separate file and have a suitable name, for example, `IStringable` (not sure if it's grammatically correct) or `ISerializableAsString` (a bit long).

Reply:
>IStringable 

Some other programming language do use that word so it is valid.
However `IStringable` from other implementations only provides `toString` method and not `fromString` method. Also I haven't found any examples of interface for mandating `fromString` method. Anyway for `fromString` method, enforcing it won't be possible now (c++17) so it would be best to revisit later after we switched to c++20.